packages/react-components/src/organisms/RelatedResearchCard.tsx

```diff
@@ -120,73 +126,107 @@ const RelatedResearchCard: React.FC<RelatedResearchCardProp> = ({
-                      researchOutputId: id,
-                    }).$
+                    'teams' in output
+                      ? sharedResearch({}).researchOutput({
+                          researchOutputId: id,
+                        }).$
+                      : gp2Routing.outputs({}).output({ outputId: id }).$
                   }
                 >
-                  {title}
+                  {outputTitle}
                 </Link>
               </p>
-              <span css={[titleStyles, rowTitleStyles]}>
-                Team or Working Group
-              </span>
-              <p css={paragraphStyle}>
-                {workingGroups?.length ? (
-                  <Link
-                    ellipsed
-                    href={
-                      network({}).workingGroups({}).workingGroup({
-                        workingGroupId: workingGroups[0].id,
-                      }).$
-                    }
-                  >
-                    {workingGroups[0].title}
-                  </Link>
-                ) : teams.length > 1 ? (
-                  'Multiple teams'
-                ) : (
-                  teams[0] && (
+              <span css={[titleStyles, rowTitleStyles]}>{tableTitles[2]}</span>
+              {'teams' in output ? (
+                <p css={paragraphStyle}>
+                  {output.workingGroups?.length ? (
                     <Link
                       ellipsed
                       href={
-                        network({}).teams({}).team({
-                          teamId: teams[0].id,
+                        network({}).workingGroups({}).workingGroup({
+                          workingGroupId: output.workingGroups[0].id,
                         }).$
                       }
                     >
-                      {teams[0].displayName}
+                      {output.workingGroups[0].title}
                     </Link>
-                  )
-                )}
-              </p>
+                  ) : output.teams.length > 1 ? (
+                    'Multiple teams'
+                  ) : (
+                    output.teams[0] && (
+                      <Link
+                        ellipsed
+                        href={
+                          network({}).teams({}).team({
+                            teamId: output.teams[0].id,
+                          }).$
+                        }
+                      >
+                        {output.teams[0].displayName}
+                      </Link>
+                    )
+                  )}
+                </p>
+              ) : (
+                output.entity &&
+                getSourceIcon && (
+                  <div css={paragraphStyle}>
+                    <div css={iconStyles}>
+                      {getSourceIcon(
+                        output.entity?.type === 'workingGroups'
+                          ? 'Working Group'
+                          : 'Project',
+                      )}
```

Comment:
We could change getSourceIcon to accept the entity.type as the property?

Reply:
committed here [review: getSourceIcon suggestion](https://github.com/yldio/asap-hub/pull/3816/commits/a7369f86600cd3b213734eb29c8e69f8bc275eb5)